code/modules/mod/modules/modules_supply.dm

```diff
@@ -174,3 +188,146 @@
+
+/obj/item/mod/module/disposal_connector/on_suit_deactivation()
+	UnregisterSignal(mod.wearer, COMSIG_MOVABLE_DISPOSING)
+
+/obj/item/mod/module/disposal_connector/get_configuration()
+	. = ..()
+	.["disposal_tag"] = add_ui_configuration("Disposal Tag", "list", GLOB.TAGGERLOCATIONS[disposal_tag], GLOB.TAGGERLOCATIONS)
+
+/obj/item/mod/module/disposal_connector/configure_edit(key, value)
+	switch(key)
+		if("disposal_tag")
+			for(var/tag in 1 to length(GLOB.TAGGERLOCATIONS))
+				if(GLOB.TAGGERLOCATIONS[tag] == value)
+					disposal_tag = tag
+					break
+
+/obj/item/mod/module/disposal_connector/proc/disposal_handling(datum/disposal_source, obj/structure/disposalholder/disposal_holder, obj/machinery/disposal/disposal_machine, hasmob)
+	SIGNAL_HANDLER
+
+	disposal_holder.destinationTag = disposal_tag
+
+/obj/item/mod/module/magnet
+	name = "MOD loader hydraulic magnet module"
+	desc = "A powerful hydraulic electromagnet able to launch crates and lockers towards the user, and keep 'em attached."
+	icon_state = "magnet_loader"
+	module_type = MODULE_ACTIVE
+	removable = FALSE
+	use_power_cost = DEFAULT_CHARGE_DRAIN*3
+	incompatible_modules = list(/obj/item/mod/module/magnet)
+	cooldown_time = 1.5 SECONDS
+	overlay_state_active = "module_magnet"
+
+/obj/item/mod/module/magnet/on_select_use(atom/target)
+	. = ..()
+	if(!.)
+		return
+	if(istype(mod.wearer.pulling, /obj/structure/closet))
+		var/obj/structure/closet/locker = mod.wearer.pulling
+		playsound(locker, 'sound/effects/gravhit.ogg', 75, TRUE)
+		locker.forceMove(mod.wearer.loc)
+		locker.throw_at(target, range = 7, speed = 4, thrower = mod.wearer)
+		return
+	if(!(target in view(mod.wearer)) || !istype(target, /obj/structure/closet))
+		balloon_alert(mod.wearer, "invalid target!")
+		return
+	var/obj/structure/closet/locker = target
+	if(locker.anchored || locker.move_resist >= MOVE_FORCE_OVERPOWERING)
+		balloon_alert(mod.wearer, "target anchored!")
+		return
+	new /obj/effect/temp_visual/mook_dust(get_turf(locker))
+	playsound(locker, 'sound/effects/gravhit.ogg', 75, TRUE)
+	locker.throw_at(mod.wearer, range = 7, speed = 3, force = MOVE_FORCE_WEAK, \
+		callback = CALLBACK(src, .proc/check_locker, locker))
+
+/obj/item/mod/module/magnet/on_deactivation(display_message = TRUE)
+	. = ..()
+	if(!.)
+		return
+	if(istype(mod.wearer.pulling, /obj/structure/closet))
+		mod.wearer.stop_pulling()
+
+/obj/item/mod/module/magnet/proc/check_locker(obj/structure/closet/locker)
+	if(!mod?.wearer)
+		return
+	if(!locker.Adjacent(mod.wearer) || !isturf(locker.loc) || !isturf(mod.wearer.loc))
+		return
+	mod.wearer.AddComponent(/datum/component/strong_pull)
+	mod.wearer.start_pulling(locker)
+	RegisterSignal(locker, COMSIG_ATOM_NO_LONGER_PULLED, .proc/on_stop_pull)
+
+/obj/item/mod/module/magnet/proc/on_stop_pull(datum/source, atom/movable/last_puller)
+	SIGNAL_HANDLER
+
+	qdel(mod.wearer.GetComponent(/datum/component/strong_pull))
```

Comment:
yuck, doesnt the component does this itself too, for the exact same signal this proc is registered to